bluesky/plans.py

```diff
@@ -2691,6 +2691,150 @@ def inner_relative_adaptive_scan():
     return (yield from inner_relative_adaptive_scan())
 
 
+def tune_centroid(
+        detectors, signal, motor, 
+        start, stop, min_step, 
+        num_points=10, 
+        step_factor=2,
+        snake=False,
+        *, md=None):
+    """
+    plan: tune a motor to the centroid of signal(motor)
+    
+    Initially, traverse the range from start to stop with
+    the number of points specified.  Repeat with progressively
+    smaller step size until the minimum step size is reached.
+    Rescans will be centered on the signal centroid
+    (for $I(x)$, centroid$= \sum{I}/\sum{x*I}$)
+    with a scan range of 2*step_factor*step of current scan.
+    
+    Set `snake=True` if your positions are reproducible
+    moving from either direction.  This will not necessarily
+    decrease the number of steps required to reach convergence.
+    Snake motion reduces the total time spent on motion
+    to reset the positioner.  For some positioners, such as 
+    those with hysteresis, snake scanning may not be appropriate.  
+    For such positioners, always approach the positions from the 
+    same direction.
+    
+    Note:  Ideally the signal has only one peak in the range to 
+    be scanned.  It is assumed the signal is not polymodal 
+    between `start` and `stop`. 
+
+    Parameters
+    ----------
+    detectors : Signal
+        list of 'readable' objects
+    signal : string
+        detector field whose output is to maximize
+    motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    start : float
+        start of range
+    stop : float
+        end of range, note: start < stop
+    min_step : float
+        smallest step size to use.
+    num_points : int, optional
+        number of points with each step size, default = 10
+    step_factor : float, optional
+        used in calculating range when 
+        maximum is found, note: step_factor > 0, default = 2
+    snake : bool, optional
+        if False (default), always scan from start to stop
+    md : dict, optional
+        metadata
+
+    Example
+    -------
+    motor = Mover('motor', {'motor': lambda x: x}, {'x': 0})
+    det = SynGauss('det', motor, 'motor', center=-1.3, Imax=1e5, sigma=0.05)
+    RE(tune_centroid([det], "det", motor, -1.5, -0.5, 0.01, 10))
+    """
+    if min_step <= 0:
+        raise ValueError("min_step must be positive")
+    if step_factor <= 0:
+        raise ValueError("step_factor must be positive")
+    if (num_points - 1) <= 2*step_factor:
+        raise ValueError(
+            "Increase num_points and/or decrease step_factor"
+            " or tune_centroid will never converge to a solution"
+        )
+    try:
+        motor_name, = motor.hints['fields']
+    except (AttributeError, ValueError):
+        motor_name = motor.name
+    _md = {'detectors': [det.name for det in detectors],
+           'motors': [motor_name],
```

Comment:
sorry for the double comment, the GH UI is being weird for me and hidding the other one as out-dated.

This should still be `motor.name` so that it can be used to look up the device in the descriptor (where the configuration is keyed on the name of the parent).